contracts/implementation/AutoClaim.sol

```diff
@@ -108,22 +110,61 @@ contract AutoClaim is IAutoClaim {
+     *                        __
+     *                       /  |
+     *    __ _ _ __ __ _ ___ `| |     _______________________________________________________________________________________________________________________
+     *   / _` | '__/ _` / __| | |    |          20 bytes          |          20 bytes         |          20 bytes          |          20 bytes         | ... |
+     *  | (_| | | | (_| \__ \_| |_   |      0th user address      |     1st user address      |      3rd user address      |      4th user address     | ... |
+     *   \__,_|_|  \__, |___/\___/    ‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾
+     *              __/ |
+     *             |___/
+     *                       ___
+     *                      |__ \
+     *    __ _ _ __ __ _ ___   ) |    _______________________________________________________________________________________________________________________
+     *   / _` | '__/ _` / __| / /    |          20 bytes          |          20 bytes         |          20 bytes          |          20 bytes         | ... |
+     *  | (_| | | | (_| \__ \/ /_    |  0th poolCommitter address | 1st poolCommitter address |  3rd poolCommitter address | 4th poolCommitter address | ... |
+     *   \__,_|_|  \__, |___/____|    ‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾
+     *              __/ |
+     *             |___/
+     *
+     * @dev The nth address in args2 should be the address of the PoolCommitter where the nth address in args1 requested an auto claim.
      */
-    function multiPaidClaimMultiplePoolCommitters(address[] calldata users, address[] calldata poolCommitterAddresses)
-        external
-        override
-    {
-        require(users.length == poolCommitterAddresses.length, "Supplied arrays must be same length");
+    function multiPaidClaimMultiplePoolCommitters(bytes calldata args1, bytes calldata args2) external override {
+        require(args1.length % 20 == 0, "args must only include addresses");
+        require(args1.length == args2.length, "args must be same length");
         uint256 reward;
-        uint256 nrUsers = users.length;
-        for (uint256 i; i < nrUsers; i++) {
-            require(poolFactory.isValidPoolCommitter(poolCommitterAddresses[i]), "Invalid pool committer contract");
-            IPoolCommitter poolCommitter = IPoolCommitter(poolCommitterAddresses[i]);
-            uint256 currentUpdateIntervalId = poolCommitter.updateIntervalId();
-            reward += claim(users[i], poolCommitterAddresses[i], poolCommitter, currentUpdateIntervalId);
+        uint256 nrUsers = args1.length / 20;
+        uint256 poolCommittersOffset;
+        uint256 userOffset;
+        uint256 currentUpdateIntervalId;
+
+        assembly {
+            poolCommittersOffset := args2.offset
+            userOffset := args1.offset
+        }
```

Comment:
Is this purely a gas golf?

Reply:
Can you get the calldata offset without inline assembly? I'm actually not sure. This was just the easiest way that presented itself when I first learned how to do it.